### ISSUE TITLE
@wdio/devtools-service udpate to 6.4.0 causes timeout at browser.reloadSession()

### ISSUE THREAD

**Issue by HananArgov** [webdriverio/webdriverio#5723]
after updating all components to 6.4.0, i've encountered an issue, that when I enable the devtools service, i'm getting a jasmine timeout, calling for `browser.reloadSession()
after some research, i've found out that the code stuck here:
https://github.com/webdriverio/webdriverio/blob/f596cd244d39a15f74dac760574e31bf2c7572d8/packages/webdriverio/src/commands/browser/reloadSession.js#L46

i've tried with version 6.3.7, and had no issue at all.

the code I used (doesn't matter if you go to a url before or not):
```
describe('simple test', () => {
  it(`Simple test verification`, () => {
    browser.reloadSession();
  });
});
```

**Comment by christian-bromann** [comment]
@HananArgov thanks for reporting, you want to take a stab at it?

**Comment by mathew-jithin** [comment]
@christian-bromann Would like to pick this up if not assigned to anyone. Will have a try :-)

**Comment by christian-bromann** [comment]
@mathew-jithin sounds good, will assign it to you then.

**Comment by mathew-jithin** [comment]
@christian-bromann Tried and I was unable to reproduce the timeout as mentioned. Is there any specific steps to reproduce the same other than that mentioned by @HananArgov ?

**Comment by christian-bromann** [comment]
Closing due to inactivity. @HananArgov please let me know if you can provide a reproducible example.

**Comment by viktor-silakov** [comment]
The same problem for me, just call reloadSession() after browser.debug() from REPL calling and get wdio hanging

**Comment by christian-bromann** [comment]
This is actually still reproducible. Any help on this would be appreciated. Given this only happens if someone used the `@wdio/devtools-service` I assume the culprit is somewhere in the [`setupHandler`](https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-devtools-service/src/index.js#L140-L181) that is being triggered if a session reloads.

**Comment by ivanixgames** [comment]
i'm seeing that _setupHandler() hangs on 
this.session = await this.target.createCDPSession();

Happens in devtools-service 6.3.7 as well.

**Comment by HananArgov** [comment]
i've taken a look at this, and maybe I found the fix. it seems that devtools-service is failing to initialize on the following error:
```
[0-0] 2020-10-06T10:15:09.467Z ERROR @wdio/utils:initialiseServices: Error: Couldn't initialise "@wdio/devtools-service".
TypeError: Cannot read property 'requestLatencyMs' of undefined
    at Object.<anonymous> (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/devtools-service/build/constants.js:53:52)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Object.require.extensions.<computed> [as .js] (/Users/hananargov/Documents/qa_automation_ts/node_modules/ts-node/src/index.ts:1032:44)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/devtools-service/build/handler/network.js:8:18)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at safeRequire (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/utils/build/utils.js:147:11)
    at initialisePlugin (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/utils/build/initialisePlugin.js:20:47)
    at initialiseServices (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/utils/build/initialiseServices.js:34:51)
    at initialiseWorkerService (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/utils/build/initialiseServices.js:82:22)
    at Runner.run (/Users/hananargov/Documents/qa_automation_ts/node_modules/@wdio/runner/build/index.js:91:40)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

i've checked and saw the constant there is mistyped profile name, which after looking at the current master, i'm seeing that it's fixed in PR #5914 

my question is when is this going to tagged and packaged?

**Comment by mgrybyk** [comment]
Hopefully today

**Comment by HananArgov** [comment]
i'm sorry to say that the update didn't fix this issue. i'm still encountering it.
to be more precise, i've followed this a bit more, and found that the issue (the timeout) occurs here:
https://github.com/webdriverio/webdriverio/blob/643b1f237c0c7e4eb0e28fca31d963d3b19a073c/packages/wdio-devtools-service/src/index.js#L150

I hope this helps in some sort

**Comment by pmerwin** [comment]
I am seeing a similar issue when trying to run performance tests ... timing out after this error =>

 (node:36753) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'promise' of undefined
    at TraceGatherer.onLoadEventFired (/Users/pmerwin/Projects/pagehealth/lib-page-health-nodejs/node_modules/@wdio/devtools-service/build/gatherer/trace.js:95:42)

**Comment by HananArgov** [comment]
made some more looking around...
I don't know much about puppeteer, but I think it has the same issue we had when using selenium, that the reloadSession doesn't actually change the target for devtools.
in `node_modules/puppeteer-core/lib/cjs/puppeteer/common/Browser.js`
i've checked the target info for the line:
```
const target = new Target_js_1.Target(targetInfo, context, () => this._connection.createSession(targetInfo), this._ignoreHTTPSErrors, this._defaultViewport);
```

I saw that `targetInfo` is actually the same.

can that be the issue?

**Comment by christian-bromann** [comment]
@HananArgov yeah I think it is. We need to reconnect the Puppeteer instance when the browser reloads.

**Comment by HananArgov** [comment]
ok, I think I may have a workaround at least... if that may cause a (very) little slowdown. I think this issue is because of the `getPuppeteer()` command.

at the beginning of the function we're checking if there's already puppeteer connected, but we don't discard it when using the `reloadSession` command, thus we are kept with the old puppeteer still there haunting us.
https://github.com/HananArgov/webdriverio/blob/7004640c23ff327b8f4e095c61f1e974337b58e5/packages/webdriverio/src/commands/browser/getPuppeteer.js#L18

i've removed the if clause with the return, and the reload works :) 

so we have 2 options:
1. the dirty option - which is to remove those 3 lines of the if clause (which will enforce on getting the puppeteer session each time, just in case something changes)
2. on reload session we'll need to clear the puppeteer session so on the dev tools will be able to catch the new one.

what do you say?

**Comment by christian-bromann** [comment]
Let's go with solution number 2

**Comment by christian-bromann** [comment]
fixed by #5993
